lib/logstash/plugin_mixins/enterprise_search/client.rb

```diff
@@ -0,0 +1,37 @@
+require 'elastic-enterprise-search'
+
+module LogStash::PluginMixins::EnterpriseSearch
```

Comment:
My concern here is on the `PluginMixin` name. A mixin is a sort of trait that can be included in another class to provide additional behaviour. In this case I think is used as "utility classes" package.
I would rename the class's path if the contained classes are not effectively mixins that can be included in other classes.

Reply:
Never mind, we have similar pattern, define a class in mixin module which is an utility/helper class, also in other plugins: https://github.com/logstash-plugins/logstash-integration-jdbc/blob/93baa6e93a009ffd338cb4fcf31a61f9288fcbeb/lib/logstash/plugin_mixins/jdbc/value_tracking.rb#L7